src/PlayerPhysics.tsx

```diff
@@ -49,18 +49,19 @@ export class PlayerPhysics {
         );
     }
 
-    rotateCells(cells: UserCell[]): UserCell[] {
+    rotateCells(cells: UserCell[], isClockwise: boolean): UserCell[] {
         console.assert(this.layout.length == this.layout[0].length);
         console.assert(this.layout.length % 2 == 1);
+        console.log(this.adjustedCells);
         const mid = Math.floor(this.layout.length / 2);
-        return cells.map(({ r, c, char, uid }) => {
+        return cells.map(({ r, c, uid, char }) => {
```

Comment:
Why reverse these?  Keep them in the same order everywhere and where you construct them.  Matters for performance a lot.